src/lex/mod.rs

```diff
@@ -318,29 +321,29 @@ impl Lexer {
         } else {
             Radix::Decimal
         };
-        let start = start as u64 - b'0' as u64;
+        let start = start as u64 - '0' as u64;
 
         // the first {digits} in the regex
         let digits = match self.parse_int(start, radix, &mut buf)? {
             Some(int) => int,
             None => {
-                if radix == Radix::Octal || radix == Radix::Decimal || self.peek() == Some(b'.') {
+                if radix == Radix::Octal || radix == Radix::Decimal || self.peek() == Some('.') {
                     start
                 } else {
                     return Err(LexError::MissingDigits(radix.try_into().unwrap()));
                 }
             }
         };
-        if self.match_next(b'.') {
+        if self.match_next('.') {
             return self.parse_float(radix, buf).map(float_literal);
         }
-        if let Some(b'e') | Some(b'E') | Some(b'p') | Some(b'P') = self.peek() {
+        if let Some('e') | Some('E') | Some('p') | Some('P') = self.peek() {
             buf.push_str(".0"); // hexf doesn't like floats without a decimal point
```

Comment:
I should probably fix this.

Reply:
This was actually written when I was still using hexf, I'm not sure if it's true for `hexponent`.